addons/sale/models/sale.py

```diff
@@ -113,6 +113,20 @@ def _compute_tax_id(self):
             order.order_line._compute_tax_id()
 
     def _search_invoice_ids(self, operator, value):
+        if operator == 'in' and value:
+            self.env.cr.execute("""
+                SELECT array_agg(so.id)
+                    FROM sale_order so
+                    JOIN sale_order_line sol ON sol.order_id = so.id
+                    JOIN sale_order_line_invoice_rel soli_rel ON soli_rel.order_line_id = sol.id
+                    JOIN account_move_line aml ON aml.id = soli_rel.invoice_line_id
+                    JOIN account_move am ON am.id = aml.move_id
+                WHERE
+                    am.type in ('out_invoice', 'out_refund') AND
+                    am.id = ANY(%s)
```

Comment:
is it the same as `am.id IN %s` ?

Reply:
The `in` can crash if the list you provide is empty (I know we check if the list is not), but yes it is more or less the same thing but safer than `in`